src/site/content/en/blog/codelab-signin-form/index.md

```diff
@@ -30,89 +32,134 @@ As you'll see, these elements enable built-in browser functionality, improve acc
-* Font-size is set for mobile and desktop.
 
-Styles are also set using the `:invalid` selector for when the email input has an invalid value.
+* `padding` and `margin` are added to inputs.
+* `font-size` is different for mobile and desktop.
+
+The `:invalid` selector is used to indicate when an input has an invalid value.
+
+The CSS layout is mobile-first:
 
-The CSS layout is mobile-first: 
-* The default CSS is for viewports less than 450 px wide.
-* The media query section sets overrides for viewports that are at least 450 px wide.
-* There are CSS variables for mobile and desktop font size.
+* The default CSS is for viewports less than `450px` wide.
+* The media query section sets overrides for viewports that are at least `450px` wide.
 
 It's very important at this point to test your code on real devices on desktop and mobile:
-* Is label and input text readable, especially for people whose vision isn't 20/20?
-* Are the inputs and **Sign in** button large enough to use as touch targets for thumbs?
 
-{% Aside 'codelab' %}
-[View complete code for Step 2](https://glitch.com/edit/#!/signin-form-codelab-2).
-{% endAside %}
+* Is label and input text readable, especially for people with low vision?
+* Are the inputs and **Log in** button large enough to use as touch targets for thumbs?
 
 ## Step 3: Add input attributes to enable built-in browser features
 
 Add attributes to your form HTML so it looks like this:
 
-```html
-<form id="signin" action="#" method="post">
-      
-  <h1>Sign in</h1>
+TODO(kayce): Is the placeholder value intentional?
+
+TODO(kayce): Clicking on the password label doesn't focus the password input.
 
+```html/3,4,7,8,10
+<form id="login" action="#" method="post">
+  <h1>Sign in</h1>
   <section>        
     <label for="email">Email</label>
     <input id="email" name="email" type="email" placeholder=" " autocomplete="email" required autofocus>
   </section>
-        
   <section>        
     <label for="current-password">Password</label>
     <input id="password" name="password" type="password" autocomplete="new-password" required>
   </section>
-
   <button id="sign-in">Sign in</button>
-  
 </form>
 ```
 
-The important parts to notice are the attributes on the label and input elements:
-*  `for="email"` and `for="current-password"`: a tap or click on the label moves focus to its input, and screenreaders announce label text when the label or the label's input gets focus.
-* `type="email"`: enable browsers to understand that the meaning of this input and, on mobile, provide a keyboard suitable for entering an email address.
-* `type="password"`: ensure browsers hide the value by default and understand that the value of this input is a password.
+Click the label that says **Email**. Notice how focus moves to the email input.
+This is because the label is associated to the input via the `for="email"` attribute.
+Screenreaders also announce the label text when the label or the label's associated
+input gets focus.
```

Comment:
This is the main thing I wanted to suggest. Rather than just describing the list of changes, we should walk the reader through experiencing firsthand how these changes are improving the form, one-by-one.

Reply:
I really like this approach: making it much more interactive rather than just asking people to absorb facts.